builtin/mainmenu/dlg_config_world.lua

```diff
@@ -160,17 +166,29 @@ local function handle_buttons(this, fields)
 		local rawlist = this.data.list:get_raw_list()
 
 		local i,mod
+		-- mod_decisions helps keeping track of what value to assign to the
+		-- load_mod_modname setting, as only one mod with a particular name
+		-- can be set.
+		-- Values: nil = not yet decided, otherwise the value of mod.enabled
+		-- for the mod that was chosen.
+		local mod_decisions = {}
 		for i,mod in ipairs(rawlist) do
+			local load_mod_setting = "load_mod_"..mod.name
 			if not mod.is_modpack and
 					not mod.is_game_content then
 				if modname_valid(mod.name) then
-					worldfile:set("load_mod_"..mod.name, tostring(mod.enabled))
+					local decision = mod_decisions[mod.name]
+					-- Is it either enabled or the first one seen?
+					if decision == nil or mod.enabled then
+						mod_decisions[mod.name] = mod.enabled
+						worldfile:set(load_mod_setting, mod.enabled and get_modpack_path(mod) or "false")
+					end
 				else
 					if mod.enabled then
-						gamedata.errormessage = fgettext_ne("Failed to enable mod \"$1\" as it contains disallowed characters. Only chararacters [a-z0-9_] are allowed.", mod.name)
+						gamedata.errormessage = fgettext_ne("Failed to enable mod \"$1\" as it contains disallowed characters. Only characters [a-z0-9_] are allowed.", mod.name)
```

Comment:
Line way too long (> 90 characters). 
Also: possible combination of `else` to `elseif mod.enabled then` to reduce indents.

Reply:
This was existing code. I just fixed a typo while I was there. I'll see what I can do about the line length.